# --- SDE-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
#
# Filename: package/.../commonc++/commoncpp2-1.8.1-statfix.patch
# Copyright (C) 2012 The OpenSDE Project
#
# More information can be found in the files COPYING and README.
#
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- SDE-COPYRIGHT-NOTE-END ---

diff -Nur commoncpp2-1.8.1.orig/src/applog.cpp commoncpp2-1.8.1/src/applog.cpp
--- commoncpp2-1.8.1.orig/src/applog.cpp	2010-10-31 17:11:55.000000000 -0600
+++ commoncpp2-1.8.1/src/applog.cpp	2012-01-22 19:23:13.639292789 -0700
@@ -45,6 +45,7 @@
 #include <cstdlib>
 #include <stdarg.h>
 #include <errno.h>
+#include <sys/stat.h>
 
 // TODO sc: test if has to move up now that it is into commoncpp
 // NOTE: the order of inclusion is important do not move following include line
@@ -297,7 +298,7 @@
     else
     {
       // create pipe
-      int err = mkfifo(_nomeFile.c_str(), S_IREAD | S_IWRITE);
+      int err = mkfifo(_nomeFile.c_str(), S_IRUSR | S_IWUSR);
       if (err == 0 || errno == EEXIST)
       {
         // and open it
@@ -342,7 +343,7 @@
     else
     {
       // create pipe
-      int err = mkfifo(_nomeFile.c_str(), S_IREAD | S_IWRITE);
+      int err = mkfifo(_nomeFile.c_str(), S_IRUSR | S_IWUSR);
       if (err == 0 || errno == EEXIST)
       {
         // and open it
@@ -456,7 +457,7 @@
     else
     {
       // create pipe
-      int err = mkfifo(d->_nomeFile.c_str(), S_IREAD | S_IWRITE);
+      int err = mkfifo(d->_nomeFile.c_str(), S_IRUSR | S_IWUSR);
       if (err == 0 || errno == EEXIST)
       {
         // and open it
@@ -562,7 +563,7 @@
     else
     {
       // create pipe
-      int err = mkfifo(d->_nomeFile.c_str(), S_IREAD | S_IWRITE);
+      int err = mkfifo(d->_nomeFile.c_str(), S_IRUSR | S_IWUSR);
       if (err == 0 || errno == EEXIST)
       {
         // and open it
